### ISSUE TITLE
Nested namespace packages: No name in module with python2.7

### ISSUE THREAD

**Issue by evanj** [PyCQA/pylint#1686]
When using pylint with python2.7, if the code imports from a nested namespace package, it returns an incorrect no-name-in-module error.

### Steps to reproduce
1. Clone latest astroid (cf5e65dcb1) and pylint (0b1b25a)
2. Create a python2 virtualenv called bugvenv2 (I used Python 2.7.12 from Ubuntu 16.04)
3. Run `setup.py install` to install astroid and pylint into the virtualenvs
4. Run the following:
```
bugvenv2/bin/pip install google-cloud-core
cat > importbug.py <<HERE
import google.cloud.exceptions
print('google.cloud.exceptions:', google.cloud.exceptions)
HERE
bugvenv2/bin/python2 importbug.py
bugvenv2/bin/pylint importbug.py
```

### Current behavior

Python will correctly import and print the module. Pylint will report:

```
E:  1, 0: No name 'cloud' in module 'google' (no-name-in-module)
E:  1, 0: Unable to import 'google.cloud.exceptions' (import-error)
I:  2,34: Module 'google' has not 'cloud' member, but source is unavailable. Consider adding this module to extension-package-whitelist if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
```

### Expected behavior

No errors: the module is able to be imported, and the source code is available. This works correctly with python3.

### pylint --version output

```
pylint 1.8.0, (installed from 0b1b25a63f)
astroid 1.5.3 (installed from cf5e65dcb1)
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]
```

### Possible cause

`google.cloud.exceptions` is nested inside two namespace packages:

* `site-packages/google` has no `__init__.py`
* `site-packages/google/cloud` has no `__init__.py`
* `site-packages/google/cloud/exceptions.py` exists
* `site-packages/google/auth/__init__.py` exists, and changing `importbug.py` to contain `import google.auth` works without detecting an error

I believe the way this works is that `google-cloud-core`, declares the following `namespace_packages` in `setup.py`: 

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/core/setup.py#L71

```
    namespace_packages=[
        'google',
        'google.cloud',
        'google.api',
    ],
```

This causes the file `site-packages/google_cloud_core-0.27.1-py3.6-nspkg.pth` to get created, which contains generated code that looks like:

```
import sys, types, os;pep420 = sys.version_info > (3, 3);p = os.path.join(sys._getframe(1).f_locals['sitedir'], *('google', 'cloud'));ie = os.path.exists(os.path.join(p,'__init__.py'));m = not ie and not pep420 and sys.modules.setdefault('google.cloud', types.ModuleType('google.cloud'));mp = (m or []) and m.__dict__.setdefault('__path__',[]);(p not in mp) and mp.append(p);m and setattr(sys.modules['google'], 'cloud', m)
```

(there are 5 lines like that in this file, one for each "namespace" part in the namespace_packages declaration from setup.py)

**Comment by Grets** [comment]
I filed a pull request https://github.com/PyCQA/astroid/pull/460 that fixes the issue for me.

**Comment by PCManticore** [comment]
This is fixed in astroid 1.6 (not yet released but I'll release it this week most likely) by @macbeth322 's PR.
